salt/minion.py

```diff
@@ -148,8 +150,14 @@ def resolve_dns(opts):
             else:
                 ret['master_ip'] = '127.0.0.1'
         except SaltSystemExit:
-            err = 'Master address: {0} could not be resolved. Invalid or unresolveable address.'.format(
-                opts.get('master', 'Unknown'))
+            unknown_str = 'unknown address'
+            master = opts.get('master', unknown_str)
+            if master == '':
+                master = unknown_str
+            if opts.get('__role') == 'syndic':
+                err = 'Master address: \'{0}\' could not be resolved. Invalid or unresolveable address. Set \'syndic_master\' value in minion config.'.format(master)
```

Comment:
Well done! From the one who introduced a lot of `{0!r}` just to get quoting for free, which will be a pain to clean up in order to move to full Unicode support. Escaped quotes! FTW!